setuptools/config.py

```diff
@@ -399,6 +401,20 @@ def parse(self):
 
             section_parser_method(section_options)
 
+    def _deprecated_config_handler(self, func, msg, warning_class):
+        """ this function will wrap around parameters that are deprecated
+        
+        :param msg: deprecation message 
+        :param warning_class: class of warning exception to be raised
+        :param func: function to be wrapped around
+        """
+        @wraps(func)
+        def config_handler(*args, **kwargs):
+            warnings.warn(msg, warning_class)
```

Comment:
Is there a `stacklevel` parameter we could supply here that would make it easier for the the user to know where the deprecated usage lies?

Reply:
I see. I am not super familiar with `pdb` but I will read up and work on this. Thank you for your comments.

Reply:
I'm not sure `stacklevel` is necessary here, but if it is, the right answer is probably 2.

The problem is that because this is a decorator, the "natural" place for the warning to be emitted is inside the function it's wrapping, which is actually a level in the stack deeper than the one where the issue is actually raised.

Considering this warning is generated in response to a keyword argument to `setup` or a keyword in `setup.cfg`, the code that causes this is not present *anywhere* in the stack. Because the `requires` keyword can be specified in either location, I don't think there's a better place for this warning.

I think the wording of the warning itself is specific enough that hopefully people will understand what it means, but if not I don't think we can fix it with `stacklevel`.